### PR TITLE
Finish Diamond wrapper

### LINKED ISSUE #6: add Diamond wrapper
This should be written into `bfilling` directory. c.f. prodigal wrapper.

### PR DESCRIPTION
**Pull request by RNAer**
Fix #6

### CODE REVIEW

File: micronota/bfillings/diamond.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        Input path for the fasta file
+    o : str
+        Output path for the formatted database file.
+    params : dict
+        Other command line parameters for diamond blastp. key is the option
+        (e.g. "-T") and value is the value for the option (e.g. "50").
+        If the option is a flag, set the value to None.
+    Returns
+    -------
+    '''
```
Comment (mortonjt) [inline micronota/bfillings/diamond.py]: Don't forget the returns

File: micronota/bfillings/diamond.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        Number of CPU cores. Default to 0, i.e. use all avaiable cores.
+    evalue : float
+        Default to 0.01. Threshold E-value.
+    params : dict
+        Other command line parameters for diamond blastp. key is the option
+        (e.g. "-T") and value is the value for the option (e.g. "50").
+        If the option is a flag, set the value to None.
+    Returns
+    -------
+    Same data type as seq with added annotation.
```
Comment (mortonjt) [inline micronota/bfillings/diamond.py]: Looks like it returns a filepath?

File: micronota/bfillings/diamond.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    diamond blastp --db db -q query.faa -o D17.faa.dmd -t tmp_dir -a
+
+    Parameters
+    ----------
+    db : str
+        The file path to diamond formatted database.
+    seq : str, skbio.Sequence, skbio.Protein, or its child classes
+        If it is a str, it must be able to read into ``skbio.Protein``.
+    cores : int.
+        Number of CPU cores. Default to 0, i.e. use all avaiable cores.
```
Comment (mortonjt) [inline micronota/bfillings/diamond.py]: avaiable -> available

File: micronota/bfillings/diamond.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+    diamond blastp --db db -q query.faa -o D17.faa.dmd -t tmp_dir -a
+
+    Parameters
+    ----------
+    db : str
+        The file path to diamond formatted database.
+    seq : str, skbio.Sequence, skbio.Protein, or its child classes
+        If it is a str, it must be able to read into ``skbio.Protein``.
+    cores : int.
```
Comment (mortonjt) [inline micronota/bfillings/diamond.py]: int. -> int

File: micronota/bfillings/diamond.py [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+    -------
+    Same data type as seq with added annotation.
+    '''
+    suffix = basename(query)
+    tmpd = mkdtemp(suffix='', prefix='diamond_', dir=out_dir)
+    daa_fp = join(out_dir, '%s.daa' % suffix)
+    if aligner == 'blastp':
+        app = DiamondBlastp
+    elif aligner == 'blastx':
+        app = DiamondBlastx
```
Comment (mortonjt) [inline micronota/bfillings/diamond.py]: Would it be a good idea to have an else clause and throw an error?

**Comment by RNAer** [comment]
Thanks, @mortonjt ! waiting for diamond conda recipes to pass tests...

**Comment by mortonjt** [comment]
Ok. Let's merge #28 first

**Comment by mortonjt** [comment]
It doesn't look like the Diamond docs or HMMER docs are being rendered by sphinx.

What do you say about merging this and fixing those docs in the next PR?

**Pull request merged**